addon-test-support/-private/async-iterator.js

```diff
@@ -88,20 +91,29 @@ export default class AsyncIterator {
   /**
    * Set a timeout to reject a promise if it doesn't get response within the timeout threshold.
    *
-   * @param {*} reject
+   * @param {*} resolve
    */
-  _setTimeout(reject) {
+  _setTimeout(resolve, reject) {
     clearTimeout(this.timeout);
     this.timer = setTimeout(() => {
       if (!this._waiting) {
         return;
       }
-      let err = new Error(
-        `EmberExam: Promise timed out after ${
+
+      if (this._emberExamExitOnError) {
+        let err = new Error(
+          `EmberExam: Promise timed out after ${
+            this._timeout
+          } s while waiting for response for ${this._request}`
+        );
+        reject(err);
+      } else {
+        // eslint-disable-next-line no-console
+        console.error(`EmberExam: Promise timed out after ${
```

Comment:
when printing, can we use `this.ui.writeLine(message, 'ERROR')` that way, all our stdout/stderr goes through the same facade.

Reply:
is `this.ui()` available on the browser? I was looking at a few projects like ember-qunit or ember-test-helpers and they use console output, so I haven't found an example to follow.

Reply:
oh shoot, no you are right. That is only in node-code for ember-addon.s